libpldmresponder/pdr_state_sensor.hpp

```diff
@@ -99,6 +103,65 @@ void generateStateSensorPDR(const DBusInterface& dBusIntf, const Json& json,
                 {
                     continue;
                 }
+                // now attach this entity to the container that was
+                // mentioned in the json, and add this entity to the
+                // parents entity assocation PDR
+
+                std::string parent_entity_path = e.value("parent_entity_path",
+                                                         "");
+                if (parent_entity_path != "" &&
+                    associatedEntityMap.contains(parent_entity_path))
+                {
+                    // find the parent node in the tree
+                    pldm_entity parent_entity =
+                        associatedEntityMap.at(parent_entity_path);
+                    pldm_entity child_entity = {pdr->entity_type,
+                                                pdr->entity_instance,
+                                                pdr->container_id};
+                    auto parent_node =
+                        pldm_entity_association_tree_find_with_locality(
+                            bmcEntityTree, &parent_entity, false);
+                    if (!parent_node)
+                    {
+                        // parent node not found in the entity association tree,
+                        // this should not be possible
+                        error(
+                            "Parent Entity of type {P_ENTITY_TYP} not found in the BMC Entity Association tree",
+                            "P_ENTITY_TYP",
+                            static_cast<unsigned>(parent_entity.entity_type));
+                        return;
+                    }
+                    pldm_entity_association_tree_add_entity(
+                        bmcEntityTree, &child_entity, pdr->entity_instance,
+                        parent_node, PLDM_ENTITY_ASSOCIAION_PHYSICAL, false,
+                        false, 0xFFFF);
+                    uint32_t bmc_record_handle = 0;
+#ifdef OEM_IBM
```

Comment:
could not substitue OEM_IBM here with oemutilshandler

Reply:
why do we need en oemutils handler here ? 

pldm_pdr_find_last_in_range() is not present under oem-ibm , its a generic function right ?